webapp/components/test-runs.html

```diff
@@ -150,6 +150,25 @@
           };
         });
       }
+
+      // These are two helper functions to encode/decode the LAST component of
+      // the test path, which may contain query strings (because of test
+      // variants), e.g. "/dom/interfaces.html?exclude=Node" <-->
+      // "/dom/interfaces.html%3Fexclude%3DNode".
+
+      encodeTestPath(path) {
+        console.assert(path.startsWith('/'));
+        let parts = path.split('/').slice(1);
+        parts.push(encodeURIComponent(parts.pop()));
+        return '/' + parts.join('/');
+      }
+
+      decodeTestPath(path) {
+        console.assert(path.startsWith('/'));
+        let parts = path.split('/').slice(1);
+        parts.push(decodeURIComponent(parts.pop()));
+        return '/' + parts.join('/');
+      }
```

Comment:
These functions can just be done on the base class; I can't imagine we'll ever decode things differently.

Reply:
Done. Also, consequently, `test-file-results{,-time-series}.html` no longer call `encodeTestPath`, but take an encoded path as a property instead.